### PR TITLE
Add support to update user attributes on login

### PR DESCRIPTION
**Pull request by lemieux**
Currently, we are only able to update the user on init. This makes possible to do it on login too. It also fixes a bug where some state was kept after logout and was not reset on login.

I also made the package pull the version from package.json instead of doing an awkward string replace operation on build.

@alavers @dannytranlx @jugarrit

### CODE REVIEW

**Comment by mspensieri** [comment]
:+1:

File: src/js/main.js [hunk 1]
```diff
@@ -116,10 +109,17 @@ _.extend(Smooch.prototype, Backbone.Events, {
 
         endpoint.sdkVersion = this.VERSION;
 
-        return this.login(options.userId, options.jwt);
+        return this.login(options.userId, options.jwt, _.pick(options, AppUser.EDITABLE_PROPERTIES));
     },
 
-    login: function(userId, jwt) {
+    login: function(userId, jwt, attributes) {
+        if (arguments.length === 2 && _.isObject(jwt)) {
```
Comment (jugarrit) [inline src/js/main.js]: Ehhh this is rather confusing to me.
Is it a big deal if we have the second param as 'null' when we only want to supply attributes?

Comment (lemieux) [inline src/js/main.js]: Yes it is for people who won't use jwt authentication.

Comment (mspensieri) [inline src/js/main.js]: @jugarrit optional arguments are awesomesauce

**Comment by jugarrit** [comment]
:+1:

File: src/js/main.js (hunk 1 above)
Comment (lemieux) [inline src/js/main.js]: ES6 makes this way easier. :)

**Pull request merged**